### PR TITLE
`config` field for `ConfigGroup` objects

### PR DESCRIPTION
**Pull request by a-alferov**

### CODE REVIEW

**Comment by a-alferov** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @a-alferov! Build has been launched!

**Review by grey-kristy — APPROVED** [review]

File: python/cm/models.py [hunk 1]
```diff
@@ -223,6 +243,45 @@ class ConfigLog(ADCMModel):
 
     __error_code__ = 'CONFIG_NOT_FOUND'
 
+    @transaction.atomic()
+    def save(self, *args, **kwargs):
```
Comment (nagarin) [inline python/cm/models.py:247]: It's hard to understand business logic behind this code, I believe unit tests would be helpful here in many ways

**Review by nagarin — CHANGES_REQUESTED** [review]
Most of all - absent unit tests

File: python/cm/adcm_config.py [hunk 2]
```diff
@@ -423,6 +425,11 @@ def ui_config(obj, cl):
             item['value'] = flat_conf[key]
         else:
             item['value'] = get_default(spec[key])
+        if flat_group_keys:
+            if spec[key].type == 'group':
+                item['group'] = any((v for k, v in flat_group_keys.items() if k.startswith(key)))
```
[the remark this answers is not in the document]
Comment (a-alferov) [reply python/cm/adcm_config.py:430]: It is assumed that the values can only be True or False

File: python/cm/models.py [hunk 3]
```diff
@@ -461,15 +520,63 @@ class ConfigGroup(ADCMModel):
...
+            if diff is None:
+                diff = {}
+            for k, v in group_keys.items():
+                if isinstance(v, Mapping):
+                    diff.setdefault(k, {})
+                    get_diff(config[k], group_keys[k], diff[k])
+                    if not diff[k]:
+                        diff.pop(k)
+                else:
+                    if v:
```
[the remark this answers is not in the document]
Comment (a-alferov) [reply python/cm/models.py:555]: Yes, but what does this code have to do with it?

File: python/cm/models.py (hunk 1 above)
Comment (a-alferov) [reply python/cm/models.py:248]: Added unit tests

File: python/cm/models.py [hunk 4]
```diff
@@ -461,15 +520,63 @@ class ConfigGroup(ADCMModel):
...
-        self.object_type.model_class().obj.get(id=self.object_id)
+        obj = self.object_type.model_class().obj.get(id=self.object_id)
+        if self._state.adding:
+            self.config = ObjectConfig.objects.create(current=0, previous=0)
+            config_log = ConfigLog.obj.get(id=obj.config.current)
+            config_log.pk = None
+            config_log.obj_ref = self.config
+            attr = config_log.attr
+            attr.update({'group_keys': self.init_group_keys(config_log.config)})
+            config_log.attr = attr
```
[the remark this answers is not in the document]
Comment (a-alferov) [reply python/cm/models.py:576]: Fixed

File: python/cm/models.py [hunk 5]
```diff
@@ -461,15 +520,63 @@ class ConfigGroup(ADCMModel):
...
+    config = models.OneToOneField(
+        ObjectConfig, on_delete=models.CASCADE, null=True, related_name='config_group'
+    )
 
     not_changeable_fields = ('id', 'object_id', 'object_type')
 
     class Meta:
         unique_together = ['object_id', 'name', 'object_type']
 
+    def init_group_keys(self, origin, group_keys=None):
```
[the remark this answers is not in the document]
Comment (a-alferov) [reply python/cm/models.py]: Fixed

**Review by nagarin — APPROVED** [review]

**Pull request closed without merging**